### PR TITLE
feat: single repo issues tab view

### PR DESCRIPTION
**Pull request by markshenouda**

### CODE REVIEW

File: solidstart-tanstackquery-tailwind-modules/src/routes/[owner]/[name]/issues.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { Switch, Match, createSignal, createEffect } from 'solid-js';
+import { createQuery } from '@tanstack/solid-query';
+import { useParams } from 'solid-start';
+import getRepoInfo from '~/services/get-repo-info';
+import { LoadingPulseDot } from '~/components/LoadingPulseDot/LoadingPulseDot';
+import { Info } from '~/types/repo-info-type';
+import { RepoHeader } from '~/components/RepoHeader';
+import { RepoAbout } from '~/components/RepoAbout';
```
Comment (SDaian) [inline solidstart-tanstackquery-tailwind-modules/src/routes/[owner]/[name]/issues.tsx:8]: it has some lint errors.. should we care about that?

**Review by SDaian — CHANGES_REQUESTED** [review]

File: solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const parseSortParams = (options, value, position) =>
```
Comment (SDaian) [inline solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/utils.tsx:1]: index.ts is empty, should we delete?

**Review by SDaian — CHANGES_REQUESTED** [review]

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      <div class="flex-shrink-0 w-1/5 text-right pr-3 flex-nowrap flex">
+        <span class="ml-2 pt-1 flex-1 flex-shrink-0">
+          {local.commentCount > 0 && (
+            <a href="#" class="">
+              <div class="flex items-center justify-end">
+                <CommentIcon class="w-5 h-5" />
+                <span class="ml-1 text-sm font-bold">{local.commentCount}</span>
+              </div>
+            </a>
+          )}
```
Comment (Megio) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx:127]: Also here we can use `<Show when=`

File: solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            closedCount={closedCount()}
+          />
+          {pageInfo() &&
+            (pageInfo().hasNextPage || pageInfo().hasPreviousPage) && (
+              <Pagination
+                tab={tabActive()}
+                pageInfo={pageInfo()}
+                owner={`${params.owner}/${params.name}/issues`}
+              />
+            )}
```
Comment (Megio) [inline solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx:103]: here we can use `<Switch><Match when=` pattern

File: solidstart-tanstackquery-tailwind-modules/src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              disabled={
+                !props.pageInfo.hasNextPage || !props.pageInfo.endCursor
+              }
+              class={styles.btn_container + ' rounded-r-lg'}
+            >
+              Next
+            </button>
+          </span>
+        </div>
+      ) : null}
```
Comment (Megio) [inline solidstart-tanstackquery-tailwind-modules/src/components/Pagination/Pagination.tsx:48]: Also here we could use `<Show when=` pattern

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+              </>
+            ) : local.state === 'OPEN' ? (
+              <PullRequestIcon class="w-5 h-5" />
+            ) : (
+              <>
+                {local.state === 'MERGED' ? (
+                  <MergedPrIcon class="w-5 h-5" />
+                ) : (
+                  <ClosedPrIcon class="w-5 h-5" />
+                )}
```
[the remark this answers is not in the document]
Comment (BigAB) [reply solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx:66]: Seems like `<Switch>` and `<Match>` would be appropriate here

**Comment by Megio** [comment]
Hi @markshenouda reviewing this PR I see that we have a lot of errors or warnings, I saw some `any` types and a `createResource` in `solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx` instead of a `createQuery`. Are you still working on it? If you need any kind of help let me know  🙂

**Review by Megio — DISMISSED** [review]

**Pull request closed without merging**